tests/framework/executeRAVENworkflow/executeRavenFromPython.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Demonstrate running RAVEN in Python workflows.
+
+  Created on Nov 3, 2021
+
+  @author: aalfonsi
+"""
+
+import os
+import sys
+
+import matplotlib.pyplot as plt
+
+# note: we use this complicated way to find RAVEN because we don't know how RAVEN
+# is installed on specific machines; it can be simplified greatly for specific applications
+frameworkDir = os.path.abspath(os.path.join(*([os.path.dirname(__file__)]+[os.pardir]*3+['framework'])))
+sys.path.append(frameworkDir)
+thisDir = os.path.abspath(os.path.dirname(__file__))
+
+frameworkTestDir = os.path.abspath(os.path.join(frameworkDir, '../tests', 'framework'))
+targetWorkflow = os.path.join(frameworkTestDir, 'test_rom_trainer.xml')
+
+# import Driver for now
+import Driver
+# import simulation
+from Simulation import Simulation as sim
+import utils.TreeStructure as TS
+# instantiate a RAVEN simulation instance
+# we simply instanciate a Simulation instance
+ravenSim = sim(frameworkDir, verbosity="all")
+# set input files (this can be encapsulated in a method in simulation (together with the XML reader)
+ravenSim.setInputFiles([targetWorkflow])
+# read tree (XML input)
+tree = TS.parse(open(targetWorkflow,'r'))
+# read entities (this part can be encapsulated in a simulation method)
+ravenSim.XMLpreprocess(tree.getroot(),os.path.dirname(os.path.abspath(targetWorkflow)))
+ravenSim.XMLread(tree.getroot(),runInfoSkip=set(["DefaultInputFile"]),xmlFilename=targetWorkflow)
+# ready to initiate the simulation
+ravenSim.initialize()
```

Comment:
Can we combine line 43-52 in a single line? This would simplify the use

Reply:
in the test case?

Reply:
Not really in the test case but in ravenSim. The goal would be to have just few methods that the user is allowed to use and have these methods that encapsulate more ravenSim methods

Reply:
This does not have to happen in this PR, just a question about the degree of freedoms we want to give to user

Reply:
Yes I agree. I did not do it because I dnk how you guys want to "finalize" this development. I would merge this and (When the design is finalized by the team) add later all the encapsulations you need.